### PR TITLE
chore: polkadot 0.9.41

### PR DESCRIPTION
**Pull request by Ad96el**
## fixes [ticket](https://github.com/KILTprotocol/kilt-node/issues/496) and fixes [ticket](https://github.com/KILTprotocol/kilt-node/issues/514)

@ntn-x2 The `RelayChainError` enum has no `ServiceError` anymore. So I reverted it to the previous state.

### Most important changes 
- Some pallets got a renaming (sc-finality-grandpa to sc-consensus-grandpa and sp-finality-grandpa to sp-consensus-grandpa). 
- `Weight::from_ref_time` and  `Weight::from_proof_size` are deprecated now. 
-  Extract syncing protocol 

All changes for version 0.9.40 can be found [here](https://github.com/paritytech/polkadot/releases/tag/v0.9.40). 

Version [0.9.41](https://github.com/paritytech/polkadot/releases/tag/v0.9.41) is a client-only release.

### CODE REVIEW

File: pallets/delegation/src/lib.rs [hunk 1]
```diff
@@ -179,7 +179,6 @@ pub mod pallet {
 	}
 
 	#[pallet::pallet]
-	#[pallet::generate_store(pub(super) trait Store)]
```
Comment (weichweich) [inline pallets/delegation/src/lib.rs:182]: The storage getter was also deprecated?

**Review by weichweich — APPROVED** [review]

File: pallets/delegation/src/lib.rs (hunk 1 above)
Comment (Ad96el) [reply pallets/delegation/src/lib.rs:182]: Yes it is [PR](https://github.com/paritytech/substrate/pull/13535)

Comment (ntn-x2) [reply pallets/delegation/src/lib.rs:182]: Storage getters seem to still be in. The PR was partially fixing the ticket, so I guess the deprecation of the storage getters will come in a next release.

**Review by ntn-x2 — APPROVED** [review]
Sehr gut! 🚀🚀🚀

**Pull request merged**